src/NodeCompiler/CompileNodeToValue.php

```diff
@@ -122,19 +118,10 @@ private function compileConstFetch(Node\Expr\ConstFetch $constNode)
      */
     private function compileClassConstFetch(Node\Expr\ClassConstFetch $node, CompilerContext $context)
     {
-        $className = \implode('\\', $node->class->parts);
+        $className = $node->class->toString();
```

Comment:
Overall, this is sane - the `NameVisitor` does indeed simplify things, but we also need to verify that it was run. Any way to do that?

Reply:
@Ocramius I've been thinking about this all day and did not find any way to verify it was run :(

Reply:
That's alright: I don't expect random folks to jump in and use custom AST nodes with this.

Reply:
Just needs documentation